aten/src/ATen/core/ivalue_inl.h

```diff
@@ -390,6 +399,30 @@ struct C10_EXPORT ivalue::Object final : c10::intrusive_ptr_target {
   std::vector<IValue> slots_;
 };
 
+// ivalue Holder that hold a PyObject*
+struct C10_EXPORT ivalue::PyObjectHolder final : c10::intrusive_ptr_target {
+ private:
+  c10::intrusive_ptr<ivalue::PyObjectHolder> intrusive_from_this() {
+    c10::raw::intrusive_ptr::incref(this); // we are creating a new pointer
+                                           // from a raw `this` pointer
+                                           // so we need to bump the refcount
+                                           // to account for this ownership
+    return c10::intrusive_ptr<ivalue::PyObjectHolder>::reclaim(this);
+  }
+ public:
+  static c10::intrusive_ptr<ivalue::PyObjectHolder> create(PyObject* py_obj) {
```

Comment:
This captures py_obj but does not call incref on it. It needs to have a name that suggests it is taking ownership of the py_obj, such as `PyObjectHolder::steal(py_obj)`

Reply:
Actually, this probably cannot go here because refcounting logic needs to happen in libtorch_python